src/hole.rs

```diff
@@ -40,6 +40,38 @@ impl HoleList {
         }
     }
 
+    /// Places a hole of a given size at a given address after a given hole
+    unsafe fn put_hole(hole: &mut Hole, addr: usize, size: usize) {
+        mem::forget(mem::replace(&mut *(addr as *mut Hole),
```

Comment:
> because Hole does not have a drop Trait?

Yes, exactly. I'm not sure why I used it in `new`… maybe `Hole` had a destructor back then and I forgot to remove it later. In any case, it shouldn't be needed here.